packages/fhir-router/src/fhirrouter.ts

```diff
@@ -265,17 +263,43 @@ export class FhirRouter extends EventTarget {
     this.router.add(method, path, handler, { interaction: interaction ?? 'operation' });
   }
 
-  find(method: HttpMethod, path: string): RouteResult<FhirRouteHandler, FhirRouteMetadata> | undefined {
-    return this.router.find(method, path);
+  find(method: HttpMethod, url: string): RouteResult<FhirRouteHandler, FhirRouteMetadata> | undefined {
+    let path = url;
+    let query: Record<string, string | string[]> | undefined;
+
+    const queryIndex = url.indexOf('?');
+    if (queryIndex >= 0) {
+      query = parseQueryString(url);
+      path = url.substring(0, queryIndex);
+    }
+
+    const result = this.router.find(method, path);
+    if (result && query) {
+      result.query = query;
+    }
+    return result;
   }
 
   async handleRequest(req: FhirRequest, repo: FhirRepository): Promise<FhirResponse> {
-    const result = this.router.find(req.method, req.pathname);
+    const url = req.url;
+    // User-specified URL is parsed into component parts, which are populated back onto the request
+    if (req.pathname) {
+      throw new OperationOutcomeError(badRequest('FhirRequest must specify url instead of pathname'));
```

Comment:
Would rejecting requests with `pathname` specified cause backwards compatibility issues? e.g. if someone were using `fhirrouter` directly without `@medplum/server`

Reply:
Possibly, but I think we may want to offer different compatibility guarantees on `fhir-router` as compared to `server`, `core`, and `mock`